### PR TITLE
Implemented the replaceProfessor method

### PR DESCRIPTION
**Pull request by Brandon-CSUN**
Implemented replaceProfessor method haves two object parameters which take in the new professor object and the section object that's being assigned to the new professor. the method is composed of different four methods:

scheduler, addProfessor, addSection, removeSection.

the first condition checks if the new professor haves any time conflicts with their daily schedule by invoking the method scheduler from the Person class. if the method returns true and then the new professor doesn't have any time conflicts with the section that's going to be added to their daily schedule. if the method returns false then there's a time conflict by trying to add the section to their daily schedule.

the second condition in the method checks if the professor object field is set to null. if set to null then just add the new professor to the section and, add the section to the new professor's daily schedule, and returns true. if not set to null then remove the section from the current professor's daily schedule, add the new professor to the section, add the section to the new professor's daily schedule, and returns true

### CODE REVIEW

**Comment by chizuo** [comment]
looks good. I don't see any logical issues to this. Going to give @harshleen8 and @Jacob-Asaad  a chance to review.

File: Models/Section.cs [hunk 1]
```diff
@@ -94,6 +94,29 @@ public string getProfessorFullName()
             return "Staff";
         }
 
+        public bool replaceProfessor(Professor prof, Section section)
+        {
+
+            if (prof.scheduler(section) == true)
+            {
+                if (professor == null)
```
Comment (harshleen8) [inline Models/Section.cs:102]: I know this is right, but as per I know its always better to write (null == professor) instead of (professor == null)

Comment (Brandon-CSUN) [reply Models/Section.cs:102]: I did that because I am evaluating the element that return from the method if they are equal to current state of field

Comment (Jacob-Asaad) [reply Models/Section.cs:102]: I think the way he wrote it makes more sense, it sounds weird checking to see something that is null to be equivalent to an object

**Comment by Jacob-Asaad** [comment]
Looks good to me, going to merge

**Pull request merged**